File: remote/src/main/java/com/kelsos/mbrc/interactors/TrackCoverInteractorImpl.java
package com.kelsos.mbrc.interactors;

import android.graphics.Bitmap;

import com.google.inject.Inject;
import com.kelsos.mbrc.repository.TrackRepository;

import rx.Observable;

public class TrackCoverInteractorImpl implements TrackCoverInteractor {
  @Inject private TrackRepository repository;
  @Override
  public Observable<Bitmap> execute(boolean reload) {
    return repository.getTrackCover(reload);
  }
}

File: mbrc-lib/src/main/kotlin/com/kelsos/mbrc/interactors/TrackCoverInteractorImpl.kt
package com.kelsos.mbrc.interactors

import android.graphics.Bitmap

import com.google.inject.Inject
import com.kelsos.mbrc.repository.TrackRepository

import rx.Observable

class TrackCoverInteractorImpl : TrackCoverInteractor {
    @Inject private lateinit var repository: TrackRepository
    override fun execute(b: Boolean): Observable<Bitmap?> {
        return repository.getTrackCover(b)
    }
}
